internal/venice-client-common/src/main/java/com/linkedin/venice/compute/ComputeUtils.java

```diff
@@ -315,17 +321,32 @@ public static GenericRecord computeResult(
     }
 
     Map<String, String> errorMap = new HashMap<>();
-    for (ComputeOperation computeOperation: operations) {
-      ReadComputeOperator operator = ComputeOperationType.valueOf(computeOperation).getOperator();
-      String errorMessage =
-          validateNullableFieldAndGetErrorMsg(operator, inputRecord, operator.getOperatorFieldName(computeOperation))
-              .orElse(null);
+    ComputeOperation computeOperation;
+    ReadComputeOperator operator;
+    String operatorFieldName, errorMessage;
+    Schema.Field operatorField, resultField;
+    for (int i = 0; i < operations.size(); i++) {
```

Comment:
What is the reason to change it to a counter-based for loop?

Reply:
I believe it was done for performance reason. But not sure how much extra these micro-optimization would yield.
https://stackoverflow.com/questions/256859/is-there-a-performance-difference-between-a-for-loop-and-a-for-each-loop

Reply:
I'm accessing two lists so I need the `i`.

Reply:
So using two list iterator would work?

Reply:
Yes it would, but does it matter...? Is two iterators that much cleaner? I'm wary of the garbage cost of allocating iterators, since this is one of our hottest paths.

Reply:
The iterator comment is mainly for defensive coding, not a big concern.
I was considering if someone changes the implementation of `operations` to use LinkedList, the for-loop will perform worse than an iterator.
In practice, it doesn't matter with the current logic.